src/presentation/device.rs

```diff
@@ -235,6 +323,21 @@ impl SessionManager {
             .collect())
     }
 
+    /// When the device is ready to respond, it prepares the response specifying the permitted items.
+    ///
+    /// It changes the internal state to [State::Signing],
+    /// and you need
+    /// to call [SessionManager::get_next_signature_payload] and then [SessionManager::submit_next_signature]
+    /// to sign the documents.
+    ///
+    /// # Example
+    ///
+    /// ```text
```

Comment:
Instead of `text` you can use [`ignore` to keep the syntax highlight but not run the doc test](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#attributes). It would be even better to define all the variables as [hidden portions of the example](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example) so the example can be tested but I understand this can be a hassle.

Reply:
Yes, some would require a lot of code in hidden.